src/main/java/org/zalando/nakadi/service/subscription/zk/CuratorZkSubscriptionClient.java

```diff
@@ -317,4 +316,48 @@ public ZkSubscriptionNode getZkSubscriptionNodeLocked() {
 
         return subscriptionNode;
     }
+
+    @Override
+    public ZKSubscription subscribeForCursorsReset(final Runnable listener) {
+        cursorResetCache = new NodeCache(curatorFramework, getSubscriptionPath("/cursor_reset"));
+        cursorResetCache.getListenable().addListener(() -> listener.run());
+
+        try {
+            cursorResetCache.start();
+        } catch (final Exception e) {
+            throw new NakadiRuntimeException(e);
+        }
+        return new ZKSubscription() {
+            @Override
+            public void refresh() {
+                try {
+                    if (cursorResetCache != null) {
+                        cursorResetCache.rebuild();
+                    }
+                } catch (final Exception e) {
+                    throw new NakadiRuntimeException(e);
+                }
+            }
+
+            @Override
+            public void cancel() {
+                try {
+                    cursorResetCache.getListenable().clear();
+                    cursorResetCache.close();
+                    cursorResetCache = null;
+                } catch (final IOException e) {
+                    throw new NakadiRuntimeException(e);
+                }
+            }
+        };
+    }
+
+    @Override
+    public boolean isCursorResetInProgress() {
+        if (cursorResetCache == null) {
+            return false;
+        }
+        return cursorResetCache.getCurrentData() != null;
+    }
```

Comment:
I'm probably stupid, but I can not get this code. 
Each time when one call subscribeForCursorsReset() there will be cursorResetCache built. 
call for isCursorResetInProgress is dependent on previous call. (and btw do not work all the time, cause StartingState is initialized before StreamingState)